src/cobra/io/web/bigg_models_repository.py

```diff
@@ -0,0 +1,84 @@
+"""Provide a concrete implementation of the BioModels repository interface."""
+
+
+from io import BytesIO
+from typing import Union
+
+import httpx
+
+from .abstract_model_repository import AbstractModelRepository
+
+
+class BiGGModels(AbstractModelRepository):
+    """
+    Define a concrete implementation of the BiGG Models repository.
+
+    Attributes
+    ----------
+    name : str
+        The name of the BiGG Models repository.
+
+    """
+
+    name: str = "BiGG Models"
+
+    def __init__(
+        self,
+        *,
+        url: Union[httpx.URL, str] = "http://bigg.ucsd.edu/static/models/",
+        **kwargs,
+    ):
+        """
+        Initialize a BiGG Models repository interface.
+
+        Parameters
+        ----------
+        url : httpx.URL or str, optional
+            The base URL from where to load the models (default
+            http://bigg.ucsd.edu/static/models/).
+
+        Other Parameters
+        ----------------
+        kwargs
+            Passed to the parent constructor in order to enable multiple inheritance.
+
+        """
+        super().__init__(url=url, **kwargs)
+
+    def get_sbml(self, model_id: str) -> bytes:
```

Comment:
There seems to be redundancy between the Biomodels and BIGG implementations. Maybe first build a general SBML DB loader and then juts set the base URL in the BIGG and Biomodels classes.

Reply:
They are very similar but there are some slight variations, for example, URL parameters, content-length header. Other databases may be also slightly different. That's why I opted for 'duplicating' the code.

Reply:
Fair enough. :+1: